llvm/lib/SandboxIR/SandboxIR.cpp

```diff
@@ -1150,6 +1150,22 @@ Value *PHINode::hasConstantValue() const {
   llvm::Value *LLVMV = cast<llvm::PHINode>(Val)->hasConstantValue();
   return LLVMV != nullptr ? Ctx.getValue(LLVMV) : nullptr;
 }
+void PHINode::replaceIncomingBlockWith (const BasicBlock *Old, BasicBlock *New) {
+  assert(New && Old && "Sandbox IR PHI node got a null basic block!");
+  for (unsigned Op = 0,
+            NumOps = cast<llvm::PHINode>(Val)->getNumOperands(); Op != NumOps; ++Op)
+    if (getIncomingBlock(Op) == Old)
+      setIncomingBlock(Op, New);
+}
+void PHINode::removeIncomingValueIf(function_ref< bool(unsigned)> Predicate,
+                                    bool DeletePHIIfEmpty) {
```

Comment:
`DeletePHIIfEmpty` is dead?

Reply:
Good catch, should be passed through to the llvmir function.

Reply:
Blah. It turns out that deleting empty PHINodes is unimplemented in the tracker right now, and needs cleanup in several places. I'll delete this use for now and fix it in a variety of places in a later follow up.